services-js/registry-certs/client/common/ReviewCertificateRequest.tsx

```diff
@@ -0,0 +1,223 @@
+/** @jsx jsx */
+
+import { css, jsx } from '@emotion/core';
+
+import { ChangeEvent, Component, ReactChild } from 'react';
+
+import Router from 'next/router';
+
+import { observer } from 'mobx-react';
+
+import { CHARLES_BLUE, SERIF } from '@cityofboston/react-fleet';
+
+import { capitalize } from '../../lib/helpers';
+
+import { CERTIFICATE_COST } from '../../lib/costs';
+
+import CostSummary from './CostSummary';
+
+import QuantityDropdown from './QuantityDropdown';
+import BackButton from './question-components/BackButton';
+
+import { THIN_BORDER_STYLE } from './question-components/styling';
+
+interface Props {
+  testDontScroll?: boolean;
+  certificateType: 'birth' | 'marriage';
+  certificateRequest: any;
+  siteAnalytics: any;
+  children: ReactChild | ReactChild[];
+}
+
+/**
+ * Component which allows a user to review their request, and update the
+ * quantity of certificates they are requesting.
+ *
+ * User can proceed to /checkout, go back to the questions flow, or
+ * clear all information and start over.
+ */
+@observer
+export default class ReviewCertificateRequest extends Component<Props> {
+  componentDidMount() {
+    const { certificateType, siteAnalytics, testDontScroll } = this.props;
+
+    if (!testDontScroll) {
+      window.scroll(0, 0);
+    }
+
+    // // Since user has provided all needed information by this point, we
+    // // will count this birth certificate as a trackable product.
+    siteAnalytics.addProduct(
+      '0',
+      `${capitalize(certificateType)} certificate`,
+      `${capitalize(certificateType)} certificate`,
+      this.props.certificateRequest.quantity,
+      CERTIFICATE_COST[certificateType.toUpperCase()] / 100
+    );
+
+    siteAnalytics.setProductAction('detail');
+  }
+
+  private handleQuantityChange = (
+    event: ChangeEvent<HTMLInputElement | HTMLSelectElement>
+  ) => {
+    const { certificateRequest, certificateType, siteAnalytics } = this.props;
+
+    const oldValue = certificateRequest.quantity;
+    // Quantity can never be less than 1
+    const newValue = +event.target.value || 1;
+    const difference = Math.abs(oldValue - newValue);
+
+    // Update quantity; if user erases value in field, quantity will return to 1
+    certificateRequest.setQuantity(newValue);
+
+    siteAnalytics.sendEvent('change certificate quantity', {
+      category: capitalize(certificateType),
+      label: oldValue > newValue ? 'decrease' : 'increase',
+      value: oldValue > newValue ? -difference : difference,
+    });
+  };
+
+  private userResetStartOver = () => {
+    const { certificateType, siteAnalytics } = this.props;
+
+    this.props.certificateRequest.clearCertificateRequest();
+
+    siteAnalytics.sendEvent('user reset', {
+      category: capitalize(certificateType),
+      label: 'start over',
+    });
+
+    Router.push(`/${certificateType}`);
```

Comment:
Be aware that this is here; you may want to update this to be explicit about the separate routes. Right now this is a hidden dependency that certificateTypes and page routes match exactly.